doc/userdoc/styleguide/styleguide.rst

```diff
@@ -0,0 +1,566 @@
+
+Writing documentation: a few tips
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Be concise - Use short words and sentences; avoid unnecessary modifiers
+
+- Be consistent - Follow this style guide and relevant templates
+
+- Be specific - Avoid vague language; clarity is key
+
+- Focus your message - Start with what's most important, and keep text relevant
+  within given section
+
+- Write for your reader - Try to understand what your reader needs to know;
+  include context to what you're saying
+
+- Provide examples - Show don't tell; if you can use code snippets, screenshots,
+  or figures to illustrate a point, do so!
+
+- Avoid terms like simply, just, or easy - your knowledge does not always equal
+  your readers.  Don't make your reader feel bad because they -simply- didn't
+  understand something.
+
+- Write positively -  Use positive language rather than negative language
+
+Remember that the primary purpose of the documentation is to provide
+information to NEST users who are looking for it.
+
+Voice
+~~~~~~
+
+Voice is *how* we talk to people through writing. The voice encompasses
+substance, style, and tone.
+
+Our voice is
+
+- empathetic - we write thinking of who we are writing for and what their needs
+  are
+- relaxed - we write in natural, conversational way rather than an authoritative
+  way
+
+Use 'you' to indicate the reader and 'we' to indicate yourself (the writer and
+possibly the NEST team).
+
+The language we use
+~~~~~~~~~~~~~~~~~~~
+
+We follow spelling and grammar rules of American English.
+
+The markup language we use is reStructuredText. This includes documentation
+blocks in files written in Python and cpp.
+
+We avoid slang or jargon that is unnecessary or could be confusing to readers.
+
+We do, however, use terminology appropriate to NEST and neuroscience when
+specific and exact terms are needed, which is often. But explanations or
+references should be provided for clarity in introductory texts.
+
+If you have further questions about style not addressed here, see <<STANDARDIZED STYLE GUIDE>>
+
+Active voice
+~~~~~~~~~~~~
+
+Prefer the active voice, where the subject acts on object.
+
+If the object needs emphasis rather than subject, use passive voice. But ensure
+that you cannot improve the sentence by using the active voice.
+
+
+Types of formatting
+-------------------
+
+We use specific formatting marks to help readers scan through and find what
+they need, as well as provide references to important terms.
+
+
+Underlines for headings
+~~~~~~~~~~~~~~~~~~~~~~~
+
+ReStructuredText uses several types of underline markers for headings. It's
+important that the length of the underline is at least as long as the words
+in the heading.
+
+In general, we try to follow the pattern of
+
+* First heading: ===
+* Second heading: ---
+* Third heading: ~~~
+* Fourth heading: ^^^
+
+"Double quotes"
+~~~~~~~~~~~~~~~
+
+We use double quotes for strings in code, for example
+(nest.Create("iaf_psc_alpha")).
```

Comment:
Yeah, I wasn't sure about this one as we have a lot of both types. 
Although consistency over all files is nice, an alternative is to say you can choose either single or double quotes, but be consistent within a single document/script. Or something like restructured text documentation files should use double quotes / single quotes.

Reply:
We faced (actually, are still facing) this issue in the NESTML codebase as well, which is a horrible incongruous mix of single and double quotes. We want to standardise on double quotes, as PEP 257 dictates the use of double quotes in triple quoted strings. Global consistency should definitely be a desideratum, but leaves one to wonder how strongly this can be enforced in practice (this will almost definitely be overlooked in PR reviews *n* years down the line).

Reply:
I added this question and the style guide as such to the "in depth" part of the next [ONDVC](https://github.com/nest/nest-simulator/wiki/2021-04-26-Open-NEST-Developer-Video-Conference).